authenticatorbridge/src/test/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerTest.kt

```diff
@@ -0,0 +1,440 @@
+
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.content.pm.PackageManager.NameNotFoundException
+import com.bitwarden.authenticatorbridge.IAuthenticatorBridgeService
+import com.bitwarden.authenticatorbridge.IAuthenticatorBridgeServiceCallback
+import com.bitwarden.authenticatorbridge.manager.model.AccountSyncState
+import com.bitwarden.authenticatorbridge.manager.model.AuthenticatorBridgeConnectionType
+import com.bitwarden.authenticatorbridge.model.EncryptedSharedAccountData
+import com.bitwarden.authenticatorbridge.model.SharedAccountData
+import com.bitwarden.authenticatorbridge.util.FakeLifecycleOwner
+import com.bitwarden.authenticatorbridge.util.FakeSymmetricKeyStorageProvider
+import com.bitwarden.authenticatorbridge.util.TestAuthenticatorBridgeCallbackProvider
+import com.bitwarden.authenticatorbridge.util.decrypt
+import com.bitwarden.authenticatorbridge.util.generateSecretKey
+import com.bitwarden.authenticatorbridge.util.toFingerprint
+import com.bitwarden.authenticatorbridge.util.toSymmetricEncryptionKeyData
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkConstructor
+import io.mockk.mockkStatic
+import io.mockk.runs
+import io.mockk.slot
+import io.mockk.unmockkConstructor
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class AuthenticatorBridgeManagerTest {
+
+    private val context = mockk<Context> {
+        every { applicationContext } returns this
+        every {
+            packageManager.getPackageInfo("com.x8bit.bitwarden.dev", 0)
+        } returns mockk()
+    }
+    private val mockBridgeService: IAuthenticatorBridgeService = mockk()
+    private val fakeLifecycleOwner = FakeLifecycleOwner()
+    private val fakeSymmetricKeyStorageProvider = FakeSymmetricKeyStorageProvider()
+    private val testAuthenticatorBridgeCallbackProvider = TestAuthenticatorBridgeCallbackProvider()
+
+    private val manager: AuthenticatorBridgeManagerImpl = AuthenticatorBridgeManagerImpl(
+        context = context,
+        connectionType = AuthenticatorBridgeConnectionType.DEV,
+        symmetricKeyStorageProvider = fakeSymmetricKeyStorageProvider,
+        callbackProvider = testAuthenticatorBridgeCallbackProvider,
+        processLifecycleOwner = fakeLifecycleOwner,
+    )
+
+    @BeforeEach
+    fun setup() {
+        mockkConstructor(Intent::class)
+        mockkStatic(IAuthenticatorBridgeService.Stub::class)
+        mockkStatic(EncryptedSharedAccountData::decrypt)
+    }
+
+    @AfterEach
+    fun teardown() {
+        unmockkConstructor(Intent::class)
+        unmockkStatic(IAuthenticatorBridgeService.Stub::class)
+        unmockkStatic(EncryptedSharedAccountData::decrypt)
+    }
+
+    @Test
+    fun `initial state should be Loading`() {
```

Comment:
```suggestion
    fun `initial state should be Loading when Bitwarden app is present`() {
```

Reply:
And actually...initial state of what? I think `accountSyncStateFlow` should probably be in the names of most or all of these functions.

Reply:
12e7c4962e5ce2ead8820f59c3a073519533599c